### PR TITLE
CI: Adding WinSxS fallback for MFC lookup

### PR DESCRIPTION
**Pull request by thopiekar**
These corrections allow to look into %WINDIR%/WinSxS for the MFC libraries and take them from there if they couldn't be found in the Visual Studio installation. That can happen when building for Python 2.7 and using the Express version of Visual Studio.

With these changes, all builds can pass now.
More importantly: We can deploy the resulting wheel files and installers directly here to Github.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -919,23 +920,35 @@ def build_extensions(self):
                 raise RuntimeError("Can't find %r" % (src,))
             self.copy_file(src, target_dir)
         else:
+            # Common values for the MFC lookup over the Visual Studio installation and redist installation.
```
[the remark this answers is not in the document]
Comment (thopiekar) [reply setup.py]: Hmm, ok. My style of coding is normally:
"Only split algorithms into functions, if you can reuse parts of it."

However, if you like some cleanup, I can see whether I can find time to do it.

File: setup.py [hunk 2]
```diff
@@ -952,10 +965,47 @@ def build_extensions(self):
...
+                windows_dir = os.getenv("windir", "C:\\Windows")
+                if os.path.isdir(windows_dir):
+                    winsxs_path = os.path.join(os.environ["windir"], "WinSxS")
+                    if os.path.isdir(winsxs_path):
+                        mfc_redist_path = None
+                        winsxs_listdir = os.listdir(winsxs_path)
+                        winsxs_listdir.sort()
+                        for entry in winsxs_listdir:
+                            if entry.startswith("{}_microsoft.{}.mfc_".format(platform.machine().lower(), mfc_version)) and os.path.isdir(os.path.join(winsxs_path, entry)):
+                                all_files_available = True
```
[the remark this answers is not in the document]
Comment (thopiekar) [reply setup.py]: Again, good catch! Thanks!

**Comment by thopiekar** [comment]
Hey @mhammond,

are you fine with the recent changes or is there anything left I need to improve?

Thanks!

File: setup.py [hunk 3]
```diff
@@ -285,7 +285,7 @@ def __init__ (self, name, sources,
                   implib_name=None,
                   delay_load_libraries="",
                  ):
-        libary_dirs = library_dirs,
+        _ = library_dirs,
```
Comment (mhammond) [inline setup.py]: It looks like this line can be removed?

**Comment by mhammond** [comment]
> More importantly: We can deploy the resulting wheel files and installers directly here to Github.

Note that I probably will not enable that without another change that somehow marks the build as "not official" - eg, marking the "build number" as something like the git hash - but that needn't block getting this merged now.

**Comment by mhammond** [comment]
Thanks!

**Comment by thopiekar** [comment]
Hey Mark,

sorry that I didn't take the time to add some lines to the CHANGES.txt. At the end of next week, there will be time for this kind of minor things, so I can add some comments then if you don't mind.

**Pull request merged**